core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -91,7 +103,8 @@
 
     // Kick off initial async jobs
     synchronized (instanceDataGuard) {
-      this.currentInstanceData = this.performRefresh();
+      this.nextInstanceData = Futures.immediateFuture(performRefresh());
```

Comment:
OK. What about this?

```java
this.currentInstanceData = performRefresh();
this.nextInstanceData = immediateFuture(currentInstanceData);
```

Now you don't need to call `blockOnNestedFuture` here.